### PR TITLE
New release process

### PR DESCRIPTION
**Pull request by hadim**
## Changelogs

- New release process:
    - Remove the requirement of `rever` for the release + remove associated files.
    - Separate documentation building for `main` and new releases.
- Update CI from `mamba-org/provision-with-micromamba@v15` to `mamba-org/setup-micromamba@v1`
- Fix test CI where pytorch and PL matrix version weren't propagated to the conda env.
- Previous changelogs before the new release process:
    - Added a dedicated section in our documentation to showcase community contributions.

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation is a new function is added, or an existing one is deleted._
- [x] _Write concise and explanatory changelogs below._
- [x] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

The old release process was using `rever` for some part and other processes for other parts + we had a couple of issues with documentation building. This PR removes totally rever, updates the CI and resolves some other issues around doc building. It also unifies the release process among the other datamol-io libs.

(maybe we'll want a bot to do that for us at some point, such as probot (https://github.com/googleapis/repo-automation-bots))

### CODE REVIEW

**Comment by hadim** [comment]
Related to https://github.com/datamol-io/datamol/pull/197

**Comment by hadim** [comment]
The only thing that is different from the previous rever process: the changelogs are only written to the GH release using the name of the PRs. Two improvements could be made here:

- Parse the detailed changelogs for every PR (looking at the first message in the PR) and add it to the changelogs below the PR's title.
- Update the `CHANGELOGS.md` file (require custom logic)

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+- ETL scripts/notebooks for transparency
+
+**Fixed:**
+
+- Nav Bar with Tabs
+- Typos in docs
+
+**Authors:**
+
+- Cas Wognum
```
Comment (maclandrol) [inline CHANGELOG.md:20]: [minor] There seems to be a mix of full name and github username.

File: docs/developers/contribute.md [hunk 2]
```diff
@@ -3,24 +3,22 @@
 Welcome to the molfeat community! We appreciate everyone's contribution and welcome all. Apart from code contributions, there are various other ways to support the community. If you would like to get involved, please see the [community documentation](../community/get_involved.md) for more details on how to contribute. However you choose to contribute, please be mindful and respect our [code of conduct](https://github.com/datamol-io/molfeat/blob/main/.github/CODE_OF_CONDUCT.md).
 
 !!! note annotate "Get inspired by community contributions"
-    As we love community contributions, we decided to create a dedicate space in our documentation to highlight and celebrate such contributions.
-    Visit [Community Contribitons](../community/contributions.md) to learn more!
+As we love community contributions, we decided to create a dedicate space in our documentation to highlight and celebrate such contributions.
```
Comment (maclandrol) [inline docs/developers/contribute.md:6]: I think this should be indented.

**Review by maclandrol — COMMENTED** [review]
Thanks @hadim, see comments

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -26,17 +27,16 @@ jobs:
...
         uses: actions/checkout@v3
 
-      - name: Setup Mamba
-        uses: mamba-org/provision-with-micromamba@v15
+      - name: Setup mamba
+        uses: mamba-org/setup-micromamba@v1
         with:
           environment-file: env.yml
-          environment-name: release
+          environment-name: my_env
```
[the remark this answers is not in the document]
Comment (hadim) [reply .github/workflows/release.yml:34]: Just using `my_env` everywhere to make copy pasting faster. The name of the env does not have any impact. I could be a random string.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -107,4 +114,13 @@ jobs:
...
           tag_name: ${{ inputs.release-version }}
-          body: "See [CHANGELOGS.rst](https://github.com/datamol-io/molfeat/blob/main/CHANGELOG.rst)."
+          body: ${{steps.github_release.outputs.changelog}}
+
+      - name: Deploy the doc
+        run: |
+          echo "Get the gh-pages branch"
+          git fetch origin gh-pages
+
+          echo "Build and deploy the doc on ${{ inputs.release-version }}"
```
[the remark this answers is not in the document]
Comment (hadim) [reply .github/workflows/release.yml:124]: Yeah, splitting the process makes the doc CI more robust.

**Review by maclandrol — APPROVED** [review]

File: docs/developers/contribute.md [hunk 5]
```diff
@@ -3,8 +3,8 @@
 Welcome to the molfeat community! We appreciate everyone's contribution and welcome all. Apart from code contributions, there are various other ways to support the community. If you would like to get involved, please see the [community documentation](../community/get_involved.md) for more details on how to contribute. However you choose to contribute, please be mindful and respect our [code of conduct](https://github.com/datamol-io/molfeat/blob/main/.github/CODE_OF_CONDUCT.md).
 
 !!! note annotate "Get inspired by community contributions"
-  As we love community contributions, we decided to create a dedicate space in our documentation to highlight and celebrate such contributions.
-  Visit [Community Contribitons](../community/contributions.md) to learn more!
+    As we love community contributions, we decided to create a dedicate space in our documentation to highlight and celebrate such contributions.
+    Visit [Community Contribitons](../community/contributions.md) to learn more!
```
Comment (hadim) [inline docs/developers/contribute.md]: Those `!!!` are great but too bad markdown editors are not aware of those... Thanks.

**Pull request merged**